2-api/cyf-ecommerce-api/server.js

```diff
@@ -11,16 +11,9 @@ const pool = new Pool({
   connectionString: process.env.PG_CONNECT,
 });
 
-// GET ALL THE CUSTOMERS
-app.get("/customers", function (req, res) {
-  pool
-    .query("SELECT * FROM customers")
-    .then((result) => res.json(result.rows))
-    .catch((error) => {
-      console.error(error);
-      res.status(500).json(error);
-    });
-});
+/////////////////////////========POST REQUESTS=========/////////////////////////////
+
+//  ADD A Customer
 app.post("/customers", function (req, res) {
   let customerName = req.body.name;
   let customerAddress = req.body.address;
```

Comment:
Remember when checking for validity, that this is a post method and the client might not provide all the details, which is okay, you need to check in the customers table in postgres, what is not null and what can be null.

So in short, you are right to check for validity on line 23, but not all of the fields have to be not null when posting

Reply:
Even on line 37, you are assuming that all these params will be provided which might not be the case necessary

Hint: for params, refer to syllabus week 3 on psql we used empty params and filled it as we go